app/controllers/api/restful_controller.rb

```diff
@@ -92,7 +92,7 @@ def parse_date_parameters
   end
 
   def page_collection(collection)
-    collection.offset(params[:from]).limit(params[:per] || default_page_size)
+    collection.offset(params[:from]).limit(params[:per].try(:to_i) || default_page_size)
```

Comment:
params,fetch(:per, default_page_size) doesn't solve this particular issue,
because our problem was that params[:per] was "5", which was getting
compared to an integer, which was blowing up. I'm not too confident in this
fix though; it's not a problem we're seeing in master at the moment, so I'm
tempted to just close this and apply this if we run into it again.

offset(nil) is the same as offset(0), so fetching params[:from] I don't
think has a benefit either, although I do agree that in general we want to
be fetching instead of calling params directly if we have default values.

On Fri, Jun 26, 2015 at 4:39 PM, Robert Guthrie <notifications@github.com>
wrote:

> In app/controllers/api/restful_controller.rb
> <https://github.com/loomio/loomio/pull/2276#discussion_r33328294>:
>
> > @@ -92,7 +92,7 @@ def parse_date_parameters
> >    end
> >
> >    def page_collection(collection)
> > -    collection.offset(params[:from]).limit(params[:per] || default_page_size)
> > +    collection.offset(params[:from]).limit(params[:per].try(:to_i) || default_page_size)
>
> I think this should be params.fetch(:per, default_page_size) Fetch is a
> great way to provide a default when the key does not exist. Also do we want
> to apply a fetch on the params[:from] so it's params.fetch(:from, 1) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2276/files#r33328294>.
>